nixos/modules/services/continuous-integration/buildkite-agent.nix

```diff
@@ -98,10 +106,12 @@ in
             token="${catOrLiteral cfg.token}"
             name="${cfg.name}"
             meta-data="${cfg.meta-data}"
-            hooks-path="${pkgs.buildkite-agent}/share/hooks"
+            hooks-path="${cfg.hooks-path}"
             build-path="/var/lib/buildkite-agent/builds"
             bootstrap-script="${pkgs.buildkite-agent}/share/bootstrap.sh"
             EOF
+
+            chmod +x ${cfg.hooks-path}/* 2>/dev/null || true # Guard against read-only paths.
```

Comment:
This makes no sense, since `cfg.hooks-path` is a read-only nix store path. Whoever provides the path should make sure scripts are executable.

Reply:
The default does indeed point to a read-only store path -- but a different value might be used -- one pointing to a non-read-only location.

I've set the default to match the old value, although I agree that the old value doesn't make a lot of sense.

Do you have a better idea for the default?

Reply:
@deepfire the default is OK, but whoever provides the path should make sure it has correct permissions. The service should just consume it.

Reply:
Done.